torch/csrc/lazy/core/shape.cpp

```diff
@@ -33,5 +33,13 @@ std::vector<Shape> convertShapes(
   return shape;
 }
 
+size_t Shape::numel() const {
+  size_t elts = dim() > 0 ? 1 : 0;
```

Comment:
I'm not sure setting `elts` to `0` is actually necessary given https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.h#L2195 doesn't do that.

Reply:
Probably safer to keep it given TensorImpl.h might have other special handlings.

Reply:
i'm not sure, but at least my way produces the behavior that's expected (tensor of size (1,0,2) reports numel()==0)
@bdhirsh any idea if i should do this differently?